agg-ast/schema_derivation/src/schema_derivation_tests/expression.rs

```diff
@@ -122,11 +122,66 @@ mod literal {
         input = r#"{ "$timestamp": { "t": 42, "i": 1 } }"#
     );
 }
+
 mod field_ref {
     use super::*;
 
     test_derive_expression_schema!(
-        variable_ref_present,
+        ref_present,
+        expected = Ok(Schema::Atomic(Atomic::Double)),
+        input = r#""$foo""#,
+        ref_schema = Schema::Atomic(Atomic::Double),
+        variables = map!()
+    );
+
+    test_derive_expression_schema!(
+        nested_ref_present,
+        expected = Ok(Schema::AnyOf(
+            set! {Schema::Missing, Schema::Atomic(Atomic::Double)}
+        )),
+        input = r#""$foo.bar""#,
+        ref_schema = Schema::Document(Document {
+            keys: map! {
+                "bar".to_string() => Schema::Atomic(Atomic::Double)
+            },
+            ..Default::default()
+        }),
+        variables = map!()
+    );
+
+    test_derive_expression_schema!(
+        ref_missing,
+        expected = Ok(Schema::Missing),
+        input = r#""$foo""#
+    );
+
+    test_derive_expression_schema!(
+        nested_ref_set_current,
+        expected = Ok(Schema::Atomic(Atomic::Double)),
+        input = r#""$foo.bar""#,
+        variables = map! {
+            "CURRENT".to_string() => Schema::Document(Document {
+                keys: map! {
+                    "foo".to_string() => Schema::Document(Document {
+                        keys: map! {
+                            "bar".to_string() => Schema::Atomic(Atomic::Double)
+                        },
+                        required: set!["bar".to_string()],
+                        ..Default::default()
+                    })
+                },
+                required: set!["foo".to_string()],
+                ..Default::default()
+            })
+        }
+    );
+}
+
+mod variable_ref {
```

Comment:
Looks like we're missing a test for just `$$CURRENT` (being equivalent to `$$ROOT`)